lib/Basic/Subprocess.cpp

```diff
@@ -882,36 +1058,32 @@ void llbuild::basic::spawnProcess(
           continue;
         }
       }
-      activeEvents |= readfds[i].events;
+      activeEvents |= readfds[i].events != 0;
     }
 
     if (control.shouldRelease()) {
-      releaseFn([
-                 &delegate, &pgrp, pid, handle, ctx,
-                 shouldCaptureOutput,
-                 outputFd=outputPipe[0],
-                 controlFd=controlPipe[0],
-                 completionFn=std::move(completionFn)
-                 ]() mutable {
-        if (shouldCaptureOutput) {
-          captureExecutedProcessOutput(delegate, outputFd, handle, ctx);
-        } else {
-          assert(outputFd == -1);
+      std::shared_ptr<ManagedDescriptor> outputFdShared
+        = std::make_shared<ManagedDescriptor>(std::move(outputPipeParentEnd));
+      std::shared_ptr<ManagedDescriptor> controlFdShared
+        = std::make_shared<ManagedDescriptor>(std::move(controlPipeParentEnd));
```

Comment:
The shared_ptrs are necessary because we disabled the copy constructor on a `ManagedDescriptor`, which makes it impossible to copy lambdas around, which makes it impossible to capture it in a lambda with a purpose of forwarding it further.